File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<string> seq;
        unordered_set<string> ops = {"+", "-", "/", "*"};
        for (string s : tokens) {
            if (ops.count(s)) {
                int second = stoi(seq.top());
                seq.pop();
                int first = stoi(seq.top());
                seq.pop();
                int result;
                if (s == "+") {
                    result = first + second;
                }
                else if (s == "-") {
                    result = first - second;
                }
                else if (s == "*") {
                    result = first * second;
                }
                else {
                    result = first / second;
                }
                seq.push(to_string(result));
            }
            else {
                seq.push(s);
            }
        }
        return stoi(seq.top());
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-2.py
class Solution:
    def countSubstrings(self, s: str) -> str:
        count = 0

        if len(s) == 1:
            return 1
        if len(s) == 2:
            if s[0] == s[1]:
                return 3
            else:
                return 2

        for i in range(0, len(s)):
            l = i
            r = i
            while l >= 0 and r < len(s):
                if s[l] == s[r]:
                    l -= 1
                    r += 1
                    count += 1
                else:
                    break
        
        for i in range(0, len(s) - 1):
            l = i
            r = i + 1
            while l >= 0 and r < len(s):
                if s[l] == s[r]:
                    l -= 1
                    r += 1
                    count += 1
                else:
                    break        
        return count